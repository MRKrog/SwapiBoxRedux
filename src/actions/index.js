
export const saveMovie = (info) => ({
  type: 'SAVE_MOVIE',
  info
})

export const savePeople = (people) => ({
  type: 'SAVE_PEOPLE',
  people
})

export const savePlanets = (planets) => ({
  type: 'SAVE_PLANETS',
  planets
})

export const saveVehicles = (vehicles) => ({
  type: 'SAVE_VEHICLES',
  vehicles
})

export const saveFavAmount = (favorites) => ({
  type: "UPDATE_COUNT",
  favorites
})
