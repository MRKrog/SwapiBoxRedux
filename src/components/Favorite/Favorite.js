import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const Favorite = ({ favAmount, viewAllFavs }) => {
  return (
    <div className="Favorites">
      <NavLink to="/Favorites">
        <button onClick={viewAllFavs}>View Favorites<span className="count">{favAmount}</span></button>
      </NavLink>
    </div>
  )
}

Favorite.propTypes = {
  favAmount: PropTypes.number,
  viewAllFavs: PropTypes.func
}

export default Favorite;
