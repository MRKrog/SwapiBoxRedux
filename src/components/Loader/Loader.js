import React from 'react';
import PropTypes from 'prop-types';

import gear from '../../images/gear-logo.png';

const Loader = ({ loading }) => {
  const loadingClass = loading ? 'loading-true' : null;
  return (
    <div className='logo-images'>
      <img src={gear} className={loadingClass} alt="logo" />
    </div>
  )
}

Loader.propTypes = {
  loading: PropTypes.bool
}

export default Loader;
