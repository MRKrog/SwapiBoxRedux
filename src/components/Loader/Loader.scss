
.logo-images {
  display: block;
  position: relative;
  margin: auto;
  img {
    height: 300px;
  }
}

.loading-true {
  animation: App-logo-spin infinite 20s linear;
  z-index: 2;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
