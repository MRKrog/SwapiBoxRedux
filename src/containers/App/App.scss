* {
  box-sizing: border-box;
}

html, body, #root, .App {
  height: 100%;
}


.MovieTitle {
  display: flex;
  justify-content: center;
  padding: 1em;
  align-items: center;
  background: #231f20;
  box-shadow: 0px 2px 10px 0px black;

  h1 {
    font-family: $font-title;
    color: #7d7d7d;
    margin: 0;
    line-height: 0;
    text-shadow: 0px 3px 15px black;
    img {
      width: 60%;
      margin-left: auto;
      display: block;
      opacity: .5;
    }
  }

  h3 {
    margin: 0;
    line-height: 0;
    padding-left: 2em;
    img {
      width: 60px;
      margin-right: auto;
      display: block;
    }
  }

  .Favorites {
    position: absolute;
    right: 2em;

    button {
      background: none;
      border: none;
      outline: none;
      font-family: $font-Label;
      font-size: 1em;
      cursor: pointer;
      color: #42413e;
      background: #8a7c50;
      line-height: 1.5;
      font-weight: 600;
      padding: .1em .5em 0 .5em;
      border-radius: 5px;
      opacity: 1;
      transition: all .5s ease;
      box-shadow: -1px 2px 11px 0px black;
      &:hover {
        opacity: .7
      }

      span {
        padding-left: 5px;
        font-weight: bold;
      }
    }
    &.btnActive {
      background:#231f20;
      pointer-events: none;
    }
  }
}


.MovieButtons {
  display: flex;
  justify-content: space-evenly;
  margin: auto;
  padding: 1em;
  background: #231f20;
  border-radius: 5px;
  max-width: 1200px;
  box-shadow: 0px 2px 10px 0px black;

  button {
    font-size: 1.2em;
    padding: 0em 1.5em;
    border-radius: 5px;
    height: 35px;
    line-height: 2;
    border: solid 1px #968d70;
    outline: none;
    color:#42413e;
    text-transform: uppercase;
    background: #8a7c50;
    font-weight: bold;
    cursor: pointer;
    font-family: $font-Label;
    opacity: 1;
    transition: all .5s ease;
    box-shadow: -1px 2px 11px 0px black;
    &:hover {
      opacity: .7
    }

    &.btnInactive {
      pointer-events: all;
    }

    &.btnActive {
      background:#231f20;
      pointer-events: none;
    }
  }
}
