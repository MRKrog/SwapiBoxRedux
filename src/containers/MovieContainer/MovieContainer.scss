
.MovieContainer {
  width: 100%;
  height: 100%;
  padding: 4em;

  .MovieInfoContainer {
    max-width: 1200px;
    position: relative;

    .MovieCardContainer {
      background: #505050bd;
      padding: 2%;
      box-shadow: inset 0px 0px 10px 1px #231f20;
      overflow: scroll;
      min-height: 500px;
      max-height: 500px;
      position: relative;

      .MovieTitleDisplay {
        position: absolute;
        z-index: 9;
        top: -0.5em;
        left: 1em;

        h3 {
          font-family: $font-Label;
          color: #231f2094;
          text-transform: uppercase;
          font-size: 5em;
        }
      }

      .MovieCardsDisplay {
        display: flex;
        flex-wrap: wrap;
        justify-content: flex-start;
        color:#fff;
        z-index: 10;
        position: relative;
        margin-top: 1.5em;
      }

      h3 {
        margin:0;
      }
    }

  }

  .shadowBottom {
    position: relative;
    z-index: 999;

    &::after {
      content:'';
      position: absolute;
      width: 100%;
      bottom: 0px;
      height: 40px;
      left: 0;
      background: linear-gradient(to bottom, rgba(76,76,76,0.1) 0%,rgba(43,43,43,0.75) 72%,rgba(19,19,19,1))
    }

  }

}
