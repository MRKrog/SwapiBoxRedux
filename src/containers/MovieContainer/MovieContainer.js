import React, { Component } from 'react';
import { connect } from 'react-redux'
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import { saveFavAmount } from '../../actions'
import Card from '../../components/Card/Card';


export class MovieContainer extends Component {
  constructor(props) {
    super(props);

  }

  handleFavBtn = (name, category) => {
    const favIndex = this.props[category].findIndex(card => card.name === name);
    this.changeFav(this.props[category][favIndex]);
    const totalFavs = this.findAllFavorites();
    this.props.storeFavAmount(totalFavs)
  }

  findAllFavorites = () => {
    const allFavorites = [...this.props.people, ...this.props.planets, ...this.props.vehicles]

    const favCards = allFavorites.filter(card => {
      return card.favorite === true;
    })
    console.log(favCards);
    return favCards.length
  }

  changeFav = (card) => {
    if(!card.favorite) {
      card.favorite = true
    } else {
      card.favorite = false
    }
  }


  render() {
    const { id } = this.props;
    let btnActive = 'btnActive';

    const cardsToDisplay = this.props.cardData.map((info, index) => (
      <Card info={info}
            handleFavBtn={this.handleFavBtn}
            key={index}
      />
    ))

    return (
      <div className="MovieContainer">
        <section className="MovieButtons">
          <NavLink to="/People">
            <button className={id === 'people' ? btnActive : ''} onClick={() => this.props.fetchPeople('people')}>People</button>
          </NavLink>
          <NavLink to="/Planets">
            <button className={id === 'planets' ? btnActive : ''} onClick={() => this.props.fetchPlanets('planets')}>Planets</button>
          </NavLink>
          <NavLink to="/Vehicles">
            <button className={id === 'vehicles' ? btnActive : ''} onClick={() => this.props.fetchVehicles('vehicles')}>Vehicles</button>
          </NavLink>
        </section>
        <div className="MovieInfoContainer">
          <section className="MovieCardContainer">
            <div className="MovieTitleDisplay"><h3>{id}</h3></div>
            <section className="MovieCardsDisplay">
            {cardsToDisplay}
            </section>
          </section>
          <section className="shadowBottom"></section>
        </div>
      </div>
    )
  }
}

MovieContainer.propTypes = {
  movieData: PropTypes.array
}

export const mapStateToProps = (state) => ({
  people: state.people,
  planets: state.planets,
  vehicles: state.vehicles,
  favAmount: state.favoriteTotal,
})

export const mapDispatchToProps = (dispatch) => ({
  storeFavAmount: (favorites) => dispatch(saveFavAmount(favorites))
})

export default connect(mapStateToProps, mapDispatchToProps)(MovieContainer);
