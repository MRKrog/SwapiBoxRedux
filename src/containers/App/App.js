import React, { Component } from 'react';
import { Route, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { saveMovie, savePeople, savePlanets, saveVehicles } from '../../actions';

import { fetchAnything } from '../../utility/fetchAnything';
import { cleanPeople, cleanPlanets, cleanVehicles } from '../../utility/fetchCleaners';
import { fetchSpecies, fetchHomeworld, getAllPlanets } from '../../utility/api';

import MovieContainer from '../MovieContainer/MovieContainer';
import MovieInfo from '../../components/MovieInfo/MovieInfo';
import Favorite from '../../components/Favorite/Favorite';
import Loader from '../../components/Loader/Loader';

import logo from '../../images/starwars_logo.png';
import emblem from '../../images/star_emblem.png';

export class App extends Component {
  constructor() {
    super();
    this.state = {
      error: '',
      loading: false,
      category: '',
      favCards: [],
    }
  }

  componentDidMount = () => {
    const movieNumber = Math.floor(Math.random() * 7) + 1
    const url = `https://swapi.co/api/films/${movieNumber}`;
    this.fetchMovie(url)
  }

  fetchMovie = async (url) => {
    try {
      const movie = await fetchAnything(url)
      this.props.storeMovie(movie)
    } catch (error) {
      this.setState({ error: error.message })
    }
  }

  fetchPeople = async (category) => {
    const url = `https://swapi.co/api/people`;

    if(!this.props.people.length) {
      try {
        this.handleLoadStart()
        const peopleData = await fetchAnything(url)
        const peopleSpecies = await fetchSpecies(peopleData.results)
        const peopleHomeworld = await fetchHomeworld(peopleSpecies)
        const allPeople = cleanPeople(peopleHomeworld)
        this.props.storePeople(allPeople);
      } catch (error) {
        this.setState({ error: error.message })
      }
    }
    this.setState({ category })
  }

  fetchPlanets = async (category) => {
    const url = `https://swapi.co/api/planets`;
    if(!this.props.planets.length) {
      this.handleLoadStart()
      try {
        const allPlanets = await fetchAnything(url)
        const allData = await getAllPlanets(allPlanets.results)
        const resolvedPlanets = cleanPlanets(allData)
        this.props.storePlanets(resolvedPlanets);
      } catch (error) {
        this.setState({ error: error.message })
      }
    }
    this.setState({ category })
  }

  fetchVehicles = async (category) => {
    const url = `https://swapi.co/api/vehicles`;
    if(!this.props.vehicles.length) {
      try {
        this.handleLoadStart()
        const vehicleData = await fetchAnything(url)
        const resolvedVehicles = await cleanVehicles(vehicleData.results)
        this.props.storeVehicles(resolvedVehicles);
      } catch (error) {
        this.setState({ error: error.message })
      }
    }
    this.setState({ category })
  }

  findAllFavorites = () => {
    const allFavorites = [...this.props.people, ...this.props.planets, ...this.props.vehicles]
    const favCards = allFavorites.filter(card => {
      return card.favorite === true;
    })
    return favCards
  }

  viewAllFavs = () => {
    const favCards = this.findAllFavorites()
    this.setState({
      category: 'favorites',
      favCards,
    })
  }

  handleLoadStart = () => {
    this.setState({
      loading: true
    })
  }

  render() {
    const { error, loading, category } = this.state;
    return (
      <div className="App">
        { error && <h2 className="error">{error}</h2> }
        <div className="MovieTitle">
          <NavLink to="/Explore">
            <h3><img src={emblem} alt="Star Wars" /></h3>
          </NavLink>
          <h1><img src={logo} alt="Star Wars" /></h1>
          <Favorite favAmount={this.props.favAmount}
                    viewAllFavs={this.viewAllFavs}
                    />
        </div>
        <Route exact path="/" render={() => {
          if(Object.keys(this.props.movie).length) {
            return <MovieInfo movieData={this.props.movie}
                              handleUser={this.handleUser} />
          } else {
            return <Loader loading={loading} />
          }
        }} />
        <Route exact path="/Explore" render={() => {
            return <MovieContainer id={'Explore'}
                                   fetchPeople={this.fetchPeople}
                                   fetchPlanets={this.fetchPlanets}
                                   fetchVehicles={this.fetchVehicles}
                                   cardData={[]} />
        }}/>
        <Route exact path="/People" render={() => {
          if(Object.keys(this.props.people).length) {
            return <MovieContainer id={category}
                                   fetchPeople={this.fetchPeople}
                                   fetchPlanets={this.fetchPlanets}
                                   fetchVehicles={this.fetchVehicles}
                                   cardData={this.props.people}
                                    />
          } else {
            return <Loader loading={loading} />
          }
        }}/>
        <Route exact path="/Planets" render={() => {
          if(Object.keys(this.props.planets).length) {
            return <MovieContainer id={category}
                                   fetchPeople={this.fetchPeople}
                                   fetchPlanets={this.fetchPlanets}
                                   fetchVehicles={this.fetchVehicles}
                                   cardData={this.props.planets} />
          } else {
            return <Loader loading={loading} />
          }
        }}/>
        <Route exact path="/Vehicles" render={() => {
          if(Object.keys(this.props.vehicles).length) {
            return <MovieContainer id={category}
                                   fetchPeople={this.fetchPeople}
                                   fetchPlanets={this.fetchPlanets}
                                   fetchVehicles={this.fetchVehicles}
                                   cardData={this.props.vehicles} />
          } else {
            return <Loader loading={loading} />
          }
        }}/>
        <Route exact path="/Favorites" render={() => {
            return <MovieContainer id={category}
                                   fetchPeople={this.fetchPeople}
                                   fetchPlanets={this.fetchPlanets}
                                   fetchVehicles={this.fetchVehicles}
                                   cardData={this.state.favCards} />
          }
        }/>
      </div>
    );
  }
}

App.propTypes = {
  storeMovie: PropTypes.func,
}

export const mapStateToProps = (state) => ({
  movie: state.movie,
  people: state.people,
  planets: state.planets,
  vehicles: state.vehicles,
  favAmount: state.favoriteTotal,
})

export const mapDispatchToProps = (dispatch) => ({
  storeMovie: (movie) => dispatch(saveMovie(movie)),
  storePeople: (people) => dispatch(savePeople(people)),
  storePlanets: (planets) => dispatch(savePlanets(planets)),
  storeVehicles: (vehicles) => dispatch(saveVehicles(vehicles)),
})

export default connect(mapStateToProps, mapDispatchToProps)(App)
