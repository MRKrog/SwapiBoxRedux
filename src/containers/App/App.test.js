import React from 'react';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import * as actions from '../../actions'
// import { saveMovie, savePeople, savePlanets, saveVehicles, saveFavAmount } from '../../actions';
import { shallow } from 'enzyme';

import MovieContainer from '../MovieContainer/MovieContainer';
import MovieInfo from '../../components/MovieInfo/MovieInfo';
import Favorite from '../../components/Favorite/Favorite';
import Loader from '../../components/Loader/Loader';

import { cleanPeople, cleanPlanets, cleanVehicles } from '../../utility/fetchCleaners';
jest.mock('../../utility/fetchCleaners')
import { fetchSpecies, fetchHomeworld, getAllPlanets } from '../../utility/api';
jest.mock('../../utility/api')

import { fetchAnything } from '../../utility/fetchAnything';
jest.mock('../../utility/fetchAnything')

const mockMovie = {title: 'episode 1', date: 'june 1970'}
const mockPeople = [{name: 'Luke', favorite: true}, {name: 'Han Solo', favorite: false}]
const mockPlanets = [{name: 'Tattoin', favorite: true}, {name: 'Hoth', favorite: false}]
const mockVehicles = [{name: 'Falcon', favorite: true}, {name: 'Tie Fighter', favorite: false}]




describe('App', () => {

  describe('App Component', () => {
    let wrapper;
    let mockUrl;

    beforeEach(() => {
      mockUrl = 'www.starwars.com';
      wrapper = shallow(<App mockMovie={{}}
                             people={[]}
                             planets={[]}
                             vehicles={[]}
                        />)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should have a default state', () => {
      wrapper = shallow(<App />, { disableLifecycleMethods: true })
      expect(wrapper.state()).toEqual({
        error: '',
        loading: false,
        category: '',
        favCards: [],
      })
    })

    it('should invoke fetchAnything when fetchMovie is called', async () => {
      await wrapper.instance().fetchMovie(mockUrl)
      expect(fetchAnything).toHaveBeenCalledWith(mockUrl)
    })

    it('should update throw an error is the url fetch is bad', async () => {
      fetchAnything.mockImplementationOnce(() => Promise.reject(new Error('Bar Url Error')))
      await wrapper.instance().fetchMovie(mockUrl)
      const expected = "Bar Url Error"

      expect(wrapper.state('error')).toEqual(expected)
    })

    it('should call fetchAnything when fetchPeople is invoked and set loading state to true', async () => {
      let category = 'newPeps';
      // let peopleWrapper = shallow(<App people={[]} />)

      await wrapper.instance().fetchPeople(category)
      expect(wrapper.state('loading')).toEqual(true)
      expect(fetchAnything).toHaveBeenCalled()
      expect(wrapper.state('category')).toEqual('newPeps')
    })

    it('should have an error added to state if the fetch fails on fetchPeople', async () => {
      let category = 'people';
      fetchAnything.mockImplementationOnce(() => Promise.reject(new Error('Bad Url')))
      await wrapper.instance().fetchPeople(category)
      expect(wrapper.state('error')).toEqual('Bad Url')
    })

    it.skip('should not call fetchAnything when fetchPeople and has people in props', async () => {
      // wont test because on componentDidMount async call is firing
      let category = 'people'
      wrapper.setProps({
        people: [{
          favorited: false,
          homeworld: 'Tatooine',
          name: 'Luke Skywalker',
          population: '200000',
          species: 'Human',
          type: 'people'
        }]
      })
      // let wrapperPeople = shallow(<App people={['name']}/>, { disableLifecycleMethods: true })

      await wrapper.instance().fetchPeople(category);
      expect(fetchAnything).not.toHaveBeenCalled()
    })

    it.skip('should call fetchSpecies, fetchHomeworld, and cleanPeople when fetchPeople is invoked', async () => {
      let category = 'people';

      await wrapper.instance().fetchPeople(category)
      expect(fetchAnything).toHaveBeenCalled()
      expect(fetchSpecies).toHaveBeenCalled()
      expect(fetchHomeworld).toHaveBeenCalled()
    })


    it('should invoke fetchAnything when fetchPlanets is called and set the state of loading and category', async () => {
      let category = 'planets'
      await wrapper.instance().fetchPlanets(category)

      expect(wrapper.state('loading')).toEqual(true)
      expect(fetchAnything).toHaveBeenCalled()
      expect(wrapper.state('category')).toEqual('planets')
    })


    it('should have an error added to state if the fetch fails on fetchPlanets', async () => {
      let category = 'planets';
      fetchAnything.mockImplementationOnce(() => Promise.reject(new Error('Bad Planets Url')))
      await wrapper.instance().fetchPeople(category)
      expect(wrapper.state('error')).toEqual('Bad Planets Url')
    })


    it('should invoke fetchAnything when fetchVehicles is called and set the state of loading and category', async () => {
      let category = 'vehicles'
      await wrapper.instance().fetchPlanets(category)

      expect(wrapper.state('loading')).toEqual(true)
      expect(fetchAnything).toHaveBeenCalled()
      expect(wrapper.state('category')).toEqual('vehicles')
    })


    it('should have an error added to state if the fetch fails on fetchVehicles', async () => {
      let category = 'vehicles';
      fetchAnything.mockImplementationOnce(() => Promise.reject(new Error('Bad Vehicles Url')))
      await wrapper.instance().fetchPeople(category)
      expect(wrapper.state('error')).toEqual('Bad Vehicles Url')
    })

    it.skip('should find all the favorited cards and return them', () => {

      const favWrapper = shallow(<App people={mockPeople} planets={mockPlanets} vehicles={mockVehicles} />)
      const expectedFavs = {name: 'Luke', favorite: true}
      // const mockPlanets = [{name: 'Tattoin', favorite: true}, {name: 'Hoth', favorite: false}]
      // const mockVehicles = [{name: 'Falcon', favorite: true}, {name: 'Tie Fighter', favorite: false}]

      expect(favWrapper.instance().viewAllFavs()).toEqual(expectedFavs)
    })

    it('should change the state of category to favorites and set the state of favCards to all favorites cards when viewAllFavs is invoked', () => {
      const favWrapper = shallow(<App people={mockPeople} planets={mockPlanets} vehicles={mockVehicles} />)
      const expectedFavs = [{name: 'Luke', favorite: true}, {name: 'Tattoin', favorite: true}, {name: 'Falcon', favorite: true}]

      favWrapper.instance().viewAllFavs()
      expect(favWrapper.state('category')).toEqual('favorites')
      expect(favWrapper.state('favCards')).toEqual(expectedFavs)

    })


    it('should change the state of loading from true to false when invoked', () => {
      expect(wrapper.state('loading')).toEqual(false)
      wrapper.instance().handleLoadStart()
      expect(wrapper.state('loading')).toEqual(true)
    })


  })


  describe('mapStateToProps', () => {

    it('should return an object with the correct properties', () => {
      const mockData = {
        movie: [],
        people: [],
        planets: [],
        vehicles: [],
        favAmount: undefined,
        category: 'random',
      }
      const expected = {
        movie: [],
        people: [],
        planets: [],
        vehicles: [],
        favAmount: undefined,
      }

      const mockProps = mapStateToProps(mockData);
      expect(mockProps).toEqual(expected);
    })

  })

  describe('mapDispatchToProps', () => {

    it('should call dispatch storeMovie', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = actions.saveMovie(mockMovie)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.storeMovie(mockMovie)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should call dispatch storePeople', () => {
      // Setup
      const mockDispatch = jest.fn();
      const actionToDispatch = actions.savePeople(mockPeople);

      // Execution
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.storePeople(mockPeople)

      // Expectations
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('shoud call dispatch storePlanets', () => {
      // Setup
      const mockDispatch = jest.fn();
      const actionToDispatch = actions.savePlanets(mockPlanets)

      // Execution
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.storePlanets(mockPlanets)

      // Expectations
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should call dispatch storeVehicles', () => {
      // Setup
      const mockDispatch = jest.fn();
      const actionToDispatch = actions.saveVehicles(mockVehicles)

      // Execution
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.storeVehicles(mockVehicles)

      // Expectations
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })


  })

})
