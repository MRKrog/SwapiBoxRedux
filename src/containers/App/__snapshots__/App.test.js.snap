// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App Component should match the snapshot 1`] = `
<div
  className="App"
>
  <h2
    className="error"
  >
    this.props.storeMovie is not a function
  </h2>
  <div
    className="MovieTitle"
  >
    <NavLink
      to="/Explore"
    >
      <h3>
        <img
          alt="Star Wars"
          src="star_emblem.png"
        />
      </h3>
    </NavLink>
    <h1>
      <img
        alt="Star Wars"
        src="starwars_logo.png"
      />
    </h1>
    <Favorite
      viewAllFavs={[Function]}
    />
  </div>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/Explore"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/People"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/Planets"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/Vehicles"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/Favorites"
    render={[Function]}
  />
</div>
`;
